taccsite_cms/static/site_cms/js/modules/wrapElements.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Create `<details>` around desired `<summary>` content and following siblings
+ *
+ * Move each content for `<summary>` into a `<summary>` of a `<details>`
+ * which surrounds it and tags after it (but before next delimiter content)
+ *
+ * @param [string] startElementSelector - How to select which elements are the start of content to wrap
+ * @param [string] endElementSelector - How to select which elements are the end of content to wrap
+ * @param [string] wrapperTagName - The name of the tag to create
+ * @param {object} [options]
+ * @param {boolean} [options.mustWrapStarter=true] - Whether to wrap "startElement" also
+ * @param {HTMLElement} [options.scopeElement=document] - The element within which to search for elements to wrap
+ * @example
+ * <script type="module">
+ *   import wrapElements from '...';
+ *   wrapElements('h3', 'hr', 'article', { wrapStarter: true });
+ * </script>
+ *
+ * // BEFORE: A heading followed by sub-headings and <hr>'s
+ * <h2>Heading</h2>
+ * <h3>Sub-Heading A</h3>
+ * <p>…</p>
+ * <img />
+ * <hr />
+ * <h3>Sub-Heading B</h3>
+ * <p>…</p>
+ * <hr />
+ *
+ * // AFTER: A heading followed by articles (that wrap sub-headings) and <hr>'s
+ * <h2>Heading</h2>
+ * <article>
+ *   <h3>Sub-Heading A</h3>
+ *   <p>…</p>
+ *   <img />
+ * </article>
+ * <hr />
+ * <article>
+ *   <h3>Sub-Heading B</h3>
+ *   <p>…</p>
+ * </article>
+ * <hr />
+ */
+export default function wrapElements(
+  startElementSelector,
+  endElementSelector,
+  wrapperTagName,
+  options = {}
+) {
+  const opts = Object.assign({
+    mustWrapStarter: true,
+    scopeElement: document
+  }, options );
+
+  if ( ! opts.scopeElement ) {
+    console.warn('The "scopeElement" options is not an element so "wrapElements" function is abandoned', opts.scopeElement );
+    return;
+  }
+
+  const starters = opts.scopeElement.querySelectorAll( startElementSelector );
+  const containerClassname = 'js-wrapElements-container';
+  const wrapperClassname = 'js-wrapElements-wrapper';
+  const starterClassname = 'js-wrapElements-starter';
+
+  starters.forEach( starter => {
+    starter.classList.add( starterClassname );
+
+    const container = starter.parentNode;
+          container.classList.add( containerClassname );
+    const wrapper = document.createElement( wrapperTagName );
+          wrapper.classList.add( wrapperClassname );
+    const ender = _getNextSibling( starter, endElementSelector );
+    const contents = _getNextSiblingsUntil( starter, endElementSelector );
+
+    container.insertBefore( wrapper, ender );
+
+    // Whether to wrap the starter element first
+    // if ( opts.mustWrapStarter ) {
+    //   wrapper.prepend( starter );
+    // }
```

Comment:
**⚠️ Post-Merge Notice of Oddity**
@wesleyboar, should this be uncommented? If so, should then be re-tested?